rcl/test/rcl/test_subscription.cpp

```diff
@@ -528,3 +528,150 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_get_options)
 
   ASSERT_EQ(NULL, rcl_subscription_get_options(nullptr));
 }
+
+/* Using bad arguments subscription methods
+ */
+TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription_bad_argument) {
+  rcl_ret_t ret = RCL_RET_OK;
+  const rosidl_message_type_support_t * ts =
+    ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BasicTypes);
+  const char * topic = "/chatter";
+  rcl_subscription_options_t subscription_options = rcl_subscription_get_default_options();
+  rcl_subscription_t subscription = rcl_get_zero_initialized_subscription();
+  rcl_node_t invalid_node = rcl_get_zero_initialized_node();
+
+  size_t pub_count = 0;
+  test_msgs__msg__BasicTypes msg;
+  test_msgs__msg__BasicTypes__init(&msg);
```

Comment:
These `_init` and `fini` functions all return codes (at least the rmw ones do), you should probably assert they return ok.

Reply:
Added for the rmw functions, didn't find examples/references for the generated messages. Let me know if for these exist as well.

Reply:
Oh, that may be best finding it with grep in the installed headers
```
grep test_msgs__msg__BasicTypes__init -R ~/ros2_master/install/test_msgs/include/test_msgs
```
In `msg/detail/basic_types__functions.h`
```
ROSIDL_GENERATOR_C_PUBLIC_test_msgs
bool
test_msgs__msg__BasicTypes__init(test_msgs__msg__BasicTypes * msg);
```

Reply:
Thanks! I'll add it

Reply:
Added with fa88846, fini in this case returns void.